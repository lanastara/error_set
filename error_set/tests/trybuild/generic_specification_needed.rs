use std::fmt::Debug;

use error_set::error_set;

error_set! {
    AuthError1<T: Debug> = {
        SourceStruct(std::fmt::Error) {},
        DoesNotExist {
            name: T,
            role: u32,
        },
        InvalidCredentials
    };
    AuthError2<T: Debug> = {
        SourceStruct(std::fmt::Error) {},
        DoesNotExist {
            name: T,
            role: u32,
        },
        InvalidCredentials
    };
    AuthError3<G: Debug> = {
        SourceStruct(std::fmt::Error) {},
        DoesNotExist {
            name: G,
            role: u32,
        },
        InvalidCredentials
    };
    LoginError = {
        IoError(std::io::Error),
    } || AuthError1 || AuthError2 || AuthError3;
}

pub fn main() {}