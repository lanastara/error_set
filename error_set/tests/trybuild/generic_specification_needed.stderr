error: A reference to AuthError1 was declared with 0 generic param(s), but the original definition takes 1.
  --> tests/trybuild/generic_specification_needed.rs:32:10
   |
32 |     } || AuthError1 || AuthError2 || AuthError3;
   |          ^^^^^^^^^^

warning: unused import: `std::fmt::Debug`
 --> tests/trybuild/generic_specification_needed.rs:1:5
  |
1 | use std::fmt::Debug;
  |     ^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
